portability-worker/src/main/java/org/dataportabilityproject/worker/PortabilityInMemoryTransferCopier.java

```diff
@@ -37,95 +37,99 @@
-    copy(exporter, importer, exportAuthData, importAuthData, emptyExportInfo);
-
-  }
-
-  /**
-   * Transfers data fropm the given {@code exporter} optionally starting at the point specified in
-   * the provided {@code exportInformation}. Imports the data using the provided {@code importer}.
-   * If there is more data to required to be exported, recursively copies using the specific {@link
-   * ExportInformation} to continue the process.
-   *
-   * @param exporter The exporter to use
-   * @param importer The importer to use
-   * @param exportAuthData The auth data for the export
-   * @param importAuthData The auth data for the import
-   * @param exportInformation Any pagination or resource information to use for subsequent calls.
-   */
-  private void copy(
-      Exporter exporter,
-      Importer importer,
-      AuthData exportAuthData,
-      AuthData importAuthData,
-      ExportInformation exportInformation) throws IOException {
-
-    logger.debug("copy iteration: {}", COPY_ITERATION_COUNTER.incrementAndGet());
-
-    // NOTE: order is important below, do the import of all the items, then do continuation
-    // then do sub resources, this ensures all parents are populated before children get
-    // processed.
-    logger.debug("Starting export, ExportInformation: {}", exportInformation);
-    ExportResult<?> exportResult = exporter.export(exportAuthData, exportInformation);
-    logger.debug("Finished export, results: {}", exportResult);
-
-    logger.debug("Starting import");
-    // TODO, use job Id?
-    importer.importItem("1", importAuthData, exportResult.getExportedData());
-    logger.debug("Finished import");
-
-    ContinuationData continuationData = (ContinuationData) exportResult.getContinuationData();
-
-    if (null != continuationData) {
-      // Process the next page of items for the resource
-      if (null != continuationData.getPaginationData()) {
-        logger.debug("start off a new copy iteration with pagination info");
-        copy(exporter, importer, exportAuthData, importAuthData,
-            new ExportInformation(continuationData.getPaginationData(),
-                exportInformation.getContainerResource()));
-      }
-
-      // Start processing sub-resources
-      if (continuationData.getContainerResources() != null && !continuationData
-          .getContainerResources().isEmpty()) {
-        for (ContainerResource resource : continuationData.getContainerResources()) {
-          copy(exporter, importer, exportAuthData, importAuthData,
-              new ExportInformation(null, resource));
-        }
-      }
+    private static final AtomicInteger COPY_ITERATION_COUNTER = new AtomicInteger();
+    private static final Logger logger = LoggerFactory
+            .getLogger(PortabilityInMemoryTransferCopier.class);
+
+    private final TransferServiceProviderRegistry registry;
+
+    public PortabilityInMemoryTransferCopier(TransferServiceProviderRegistry registry) {
```

Comment:
this one is mostly indent changes right? is the new ctor the only change?

Reply:
Yes, sorry, that's the only change.

Reply:
ok cool, thanks